### PR TITLE
code-coverage OP#40949

### PR DESCRIPTION
**Pull request by individual-it**
run coverage analysis after test runs, report it in github comment and fail CI if the coverage is below current percentage

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -6,9 +6,15 @@
 		 xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
 	<coverage>
 		<report>
-			<text outputFile="php://stdout" showUncoveredFiles="true"/>
+			<cobertura outputFile="coverage/php/cobertura.xml"/>
+			<html outputDirectory="coverage/php/" lowUpperBound="50" highLowerBound="90"/>
```
Comment (individual-it) [inline phpunit.xml:10]: for human readable inspection offline

File: phpunit.xml [hunk 2]
```diff
@@ -6,9 +6,15 @@
...
+			<text outputFile="php://stdout" showUncoveredFiles="false" showOnlySummary="true"/>
```
Comment (individual-it) [inline phpunit.xml:11]: direct text output after running the tests

File: jest.config.js [hunk 3]
```diff
@@ -6,5 +6,10 @@ module.exports = {
 	moduleNameMapper: {
 		'\\.(scss)$': '<rootDir>/tests/jest/stubs/empty.js',
 	},
-	preset: '@vue/cli-plugin-unit-jest/presets/no-babel'
+	preset: '@vue/cli-plugin-unit-jest/presets/no-babel',
+	collectCoverage: true,
+	collectCoverageFrom: ['./src/**'],
+	coverageDirectory: '<rootDir>/coverage/jest/',
+	coverageReporters: ['lcov', 'html', 'text']
```
Comment (individual-it) [inline jest.config.js:13]: lcov for automatic reports, html for human readable analysis offline, text for instant output after test run

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -45,10 +45,58 @@ jobs:
...
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          lcov-file: ./coverage/php/lcov.info
+          delete-old-comments: true
+          title: "PHP Code Coverage"
+
+      - name: JS coverage check
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: VeryGoodOpenSource/very_good_coverage@v1.2.0
+        with:
+          min_coverage: '24'
```
Comment (individual-it) [inline .github/workflows/ci.yml:94]: roughly current coverage rate

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -45,10 +45,58 @@ jobs:
...
+          path: './coverage/jest/lcov.info'
+
+      - name: PHP coverage check
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: VeryGoodOpenSource/very_good_coverage@v1.2.0
+        with:
+          min_coverage: '36'
```
Comment (individual-it) [inline .github/workflows/ci.yml:101]: roughly current coverage rate

**Review by kiranparajuli589 — APPROVED** [review]

**Pull request merged**